crates/hdk_derive/src/util.rs

```diff
@@ -12,30 +12,24 @@ pub fn to_snake_case(name: Option<String>, v_ident: &syn::Ident) -> String {
 
 pub fn ignore_enum_data(fields: &Fields) -> proc_macro2::TokenStream {
     match fields {
-        syn::Fields::Named(_) => quote::quote! {{..}},
-        syn::Fields::Unit => quote::quote! {},
-        syn::Fields::Unnamed(_) => quote::quote! {(_)},
+        Fields::Named(_) => quote::quote! {{..}},
+        Fields::Unit => quote::quote! {},
+        Fields::Unnamed(_) => quote::quote! {(_)},
     }
 }
 
 pub fn get_unit_ident(attrs: &[syn::Attribute]) -> syn::Ident {
     attrs
         .iter()
         .find(|a| {
-            a.path
+            a.path()
                 .segments
                 .last()
                 .is_some_and(|s| s.ident == "unit_enum")
         })
-        .and_then(|a| darling::util::parse_attribute_to_meta_list(a).ok())
-        .and_then(|syn::MetaList { path, nested, .. }| {
-            nested
-                .first()
-                .filter(|_| path.is_ident("unit_enum"))
-                .and_then(|f| match f {
-                    syn::NestedMeta::Meta(syn::Meta::Path(path)) => path.get_ident().cloned(),
-                    _ => None,
-                })
+        .and_then(|a| {
+            a.parse_args_with(|input: syn::parse::ParseStream| input.parse::<syn::Ident>())
+                .ok()
```

Comment:
Its now easier to parse attributes with `syn` so changed it to parse directly to the expected token.